src/routes/postbox.ts

```diff
@@ -0,0 +1,101 @@
+import uuid from "node-uuid";
+import moment from "moment";
+import path from "path";
+
+import * as log from "../logger";
+import { getPerson, savePerson } from "../db";
+import { triggerWebhook } from "../helpers/webhooks";
+import {
+  MockPerson,
+  PostboxItemEvent,
+  PostboxOwnerType,
+  PostboxDocumentType,
+} from "../helpers/types";
+import assert from "assert";
+
+const POSTBOX_ITEM_EXAMPLE = {
+  id: "d347d967ae8c4d58b93e6698b386cae9pbxi",
+  belongs_to: "3e0b990bb0f49eb1a43904e78461c0cbcper",
+  owner_type: PostboxOwnerType.PERSON,
+  created_at: "2022-01-04T13:45:05Z",
+  document_date: "2021-06-30",
+  document_type: PostboxDocumentType.BALANCE_CONFIRMATION,
+  name: "Item's name",
+  description: "Description",
+  customer_notification: true,
+  customer_confirmation: false,
+  document_size: 1667317,
+  document_content_type: "application/pdf",
+};
+
+export const createPostboxItem = async ({
+  personId,
+  name,
+  description,
+  documentType,
+  ownerType,
+}) => {
+  const person = await getPerson(personId);
+
+  const today = moment().format("YYYY-MM-DD");
+
+  if (!person.postboxItems) {
+    person.postboxItems = [];
+  }
+
+  const postboxItem = {
+    ...POSTBOX_ITEM_EXAMPLE,
+    id: uuid.v4(),
+    belongs_to: personId,
+    owner_type: ownerType,
+    created_at: today,
+    document_type: documentType,
+    name,
+    description,
+  };
+
+  person.postboxItems.push(postboxItem);
+
+  await savePerson(person);
+  return { person, postboxItem };
+};
+
+export const createPostboxItemRequestHandler = async (req, res) => {
+  const { person_id: personId } = req.params;
+
+  log.info("createPostboxItemRequestHandler()", {
+    reqBody: req.body,
+    reqParams: req.params,
+  });
+
+  const { person, postboxItem } = await createPostboxItem({
+    personId,
+    ...req.body,
+  });
+
+  await triggerWebhook({
+    type: PostboxItemEvent.POSTBOX_ITEM_CREATED,
+    payload: postboxItem,
+    personId: person.id,
+  });
+
+  res.redirect("back");
+};
+
+export const listPostboxItems = async (req, res) => {
+  const personId = req.params.person_id;
+  assert(personId, "person ID not found");
+
+  log.info(`listPostboxItems() get list of postbox items for ${personId}`);
+
+  const person = await getPerson(personId);
+  res.status(200).send(person.postboxItems);
+};
+
+export const downloadPostboxItem = async (req, res) => {
+  log.info(
+    `downloadPostboxItem() download postbox items ${req.params.postbox_item_id}`
+  );
+
+  res.download(path.join(__dirname, "../assets/sample.pdf"));
```

Comment:
let's add 404 in case when `postbox_item_id` is missing

Reply:
done :+1: https://github.com/kontist/mock-solaris/pull/79/commits/0eb2a4c9f59b959b88741c19d052e7a3dd5c56cf